internal/controller/serverset/serverset.go

```diff
@@ -176,6 +181,40 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	}, nil
 }
 
+func didNrOfReplicasChange(cr *v1alpha1.ServerSet, replicas []v1alpha1.Server) bool {
+	return len(replicas) != cr.Status.AtProvider.Replicas
+}
+
+func (e *external) populateCRStatus(cr *v1alpha1.ServerSet, serverSetReplicas []v1alpha1.Server) {
+	if cr.Status.AtProvider.ReplicaStatuses == nil || didNrOfReplicasChange(cr, serverSetReplicas) {
+		cr.Status.AtProvider.ReplicaStatuses = make([]v1alpha1.ServerSetReplicaStatus, cr.Spec.ForProvider.Replicas)
+	}
+	cr.Status.AtProvider.Replicas = len(serverSetReplicas)
+
+	for i, replica := range serverSetReplicas {
+		replicaStatus := computeStatus(serverSetReplicas[i].Status.AtProvider.State)
+		cr.Status.AtProvider.ReplicaStatuses[i] = v1alpha1.ServerSetReplicaStatus{
+			Name:         replica.Name,
+			Status:       replicaStatus,
+			ErrorMessage: "",
+			LastModified: metav1.Now(),
+		}
+	}
+}
+
+func computeStatus(state string) string {
+	// At the moment we compute the status of the Server contained in the ServerSet
+	// based on the status of the Server. However, this status does not mean
+	// That the NAS software is up and running on it.
```

Comment:
maybe we shouldn't reference NAS here, since this might be used by people not warning to use NAS at all with it

Reply:
done